src/jira/util/jwt.ts

```diff
@@ -193,7 +206,7 @@ export const verifyAsymmetricJwtTokenMiddleware = async (req: Request, res: Resp
 			return;
 		}
 
-		const publicKey = await queryAtlassianConnectPublicKey(getKeyId(token));
+		const publicKey = await queryAtlassianConnectPublicKey(getKeyId(token), isStagingTenant(req));
```

Comment:
I'd prefer if the `isStagingTenant` function didn't have to assume that that `req.body.baseUrl` exists as it's not really it's concern.  Just doing `isStagingTenant(req.body?.baseUrl)` should do.

Reply:
It already works without baseUrl in the body, in most unit tests it is not there